iothub/device/samples/how to guides/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -1,43 +1,46 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.Azure.Devices.Client.Exceptions;
-using Microsoft.Azure.Devices.Shared;
-using Microsoft.Extensions.Logging;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Azure.Devices.Client.Samples
 {
     public class DeviceReconnectionSample
     {
-        private const int TemperatureThreshold = 30;
         private static readonly Random s_randomGenerator = new();
         private static readonly TimeSpan s_sleepDuration = TimeSpan.FromSeconds(15);
 
-        private readonly SemaphoreSlim _initSemaphore = new(1, 1);
+        private static readonly SemaphoreSlim s_initSemaphore = new(1, 1);
         private readonly List<string> _deviceConnectionStrings;
         private readonly TransportType _transportType;
-        private readonly ClientOptions _clientOptions = new() { SdkAssignsMessageId = SdkAssignsMessageId.WhenUnset };
+        private static readonly ClientOptions s_clientOptions = new() { SdkAssignsMessageId = SdkAssignsMessageId.WhenUnset };
+
+        private readonly ILogger _logger;
 
         // An UnauthorizedException is handled in the connection status change handler through its corresponding status change event.
         // We will ignore this exception when thrown by client API operations.
-        private readonly Dictionary<Type, string> _exceptionsToBeIgnored = new()
+        private static readonly HashSet<Type> s_exceptionsToBeRetried = new()
         {
-            { typeof(UnauthorizedException), "Unauthorized exceptions are handled by the ConnectionStatusChangeHandler." }
+            // Unauthorized exception conditions are handled by the ConnectionStatusChangeHandler in this sample and the sample will try
+            // to reconnect indefinitely, so don't give up on an operation that sees this exception.
+            typeof(UnauthorizedException),
```

Comment:
Do we need to start checking for the status code on the IotHubClientException here instead?

Reply:
Can you elaborate?

Reply:
These are "additional" exceptions that will be retried in this sample. The app's retry policy already retries based on IotHubClientException.Transient.